operator/internal/manifests/config.go

```diff
@@ -103,6 +103,15 @@ func ConfigOptions(opt Options) config.Options {
 		protocol = "https"
 	}
 
+	// nolint:staticcheck
+	if (opt.Stack.Replication == nil || opt.Stack.Replication.Factor == 0) && opt.Stack.ReplicationFactor > 0 {
```

Comment:
nit: it's not entirely clear to me what's the goal of this part of the code, so feel like it would be a good idea to have a high-level comment about it. WDYT?
```suggestion
	// While ReplicationFactor it's not removed if a user still has it configured the operator will use its
	// value to set up the new Replication.Factor field.
	if (opt.Stack.Replication == nil || opt.Stack.Replication.Factor == 0) && opt.Stack.ReplicationFactor > 0 {
```

Reply:
This code handles the deprecation of `opt.Stack.ReplicationFactor`.